lmdeploy/pytorch/engine/engine.py

```diff
@@ -417,32 +418,30 @@ def __get_vlm_embeddings():
             return (input_embeddings, input_embedding_indexing,
                     input_embedding_ranges)
 
-        # for vlm
-        vision_embedding_inputs = None
-        if self.model_config.task_type == 'vlm':
-            history_image_nums = None
-            history_image_token_lengths = None
-            # only for cogvlm
-            if self.model_config.model_arch == 'CogVLMForCausalLM':
-                (history_image_nums,
-                 history_image_token_lengths) = __get_cogvlm_image_info()
-
-            input_embeddings = None
-            input_embedding_indexing = None
-            input_embedding_ranges = None
-            has_embedding = any(
-                [len(msg.input_embeddings) > 0 for msg in messages])
-            if has_embedding:
-                (input_embeddings, input_embedding_indexing,
-                 input_embedding_ranges) = __get_vlm_embeddings()
-
-            vision_embedding_inputs = VisionModelInputs(
-                history_lengths=history_lengths,
-                history_image_nums=history_image_nums,
-                history_image_token_lengths=history_image_token_lengths,
-                input_embeddings=input_embeddings,
-                input_embedding_indexing=input_embedding_indexing,
-                input_embedding_ranges=input_embedding_ranges)
+        # for inputs with embeddings
+        history_image_nums = None
+        history_image_token_lengths = None
+        # only for cogvlm
+        if self.model_config.model_arch == 'CogVLMForCausalLM':
```

Comment:
why remove task type check

Reply:
because for llm models, it's also could pass embeddings when calling `engine.decode` method

Reply:
I would rather add a static condition to manually enable passing embedding. Build a dataclass is heavy even when all fields are None.